dot/rpc/modules/author_test.go

```diff
@@ -228,24 +201,26 @@ func TestAuthorModule_SubmitExtrinsic(t *testing.T) {
 			},
 			args: args{
 				req: &Extrinsic{fmt.Sprintf("0x%x", testExt)},
-				res: new(ExtrinsicHashResponse),
 			},
 			wantRes: ExtrinsicHashResponse(types.Extrinsic(testExt).Hash().String()),
 		},
 	}
 	for _, tt := range tests {
+		tt.args.res = ExtrinsicHashResponse("")
```

Comment:
nit don't define it here or in the test cases (unless it's generated)

Instead define it above where it's called, i.e.

```go
			response := ExtrinsicHashResponse("")
			err := am.SubmitExtrinsic(tt.args.r, tt.args.req, &response)
```

You should only have inputs and expected data in your test cases, and this one is not really a variable input it seems. Or define it for each test case, but defining things before the subtest and after the test cases looks strange imo